frontend/src/app/study-inquiry/study-inquiry.component.ts

```diff
@@ -27,28 +27,37 @@ export class StudyInquiryComponent implements OnInit {
     private googleAnalytics: GoogleAnalyticsService
   ) {
     this.authenticationService.currentUser.subscribe(x => this.currentUser = x);
+  }
+
+  ngOnInit() {
+    this.refreshInquiriesAndParticipants();
+  }
+
+  refreshInquiriesAndParticipants() {
     if (this.currentUser) {
-      this.haveUserContact = this.currentUser.checkContact();
-      this.api.getUserStudyInquiries(this.currentUser.id).subscribe(
-        userStudyInquiries => {
-          for ( let i in userStudyInquiries ) {
-            if (this.study.id == userStudyInquiries[i].study_id) {
-              this.alreadyInquired = true;
+      this.api.getUser(this.currentUser.id).subscribe( u => {
```

Comment:
WHEW.  I suspect this fixes some sort of bug, but goodness that is a lot of API queries you are firing off there. Do you really need to get the user afresh?  Why not use the authentication service to get the current user.  Looping through all all the particpants and calling get flow to get the participants percent complete - this all feels like a broken API to me.  I'd strongly consider re-working the api to provide this information in a sensible way.